gpus/crosstool/clang/bin/crosstool_wrapper_driver_is_not_gcc.tpl

```diff
@@ -56,6 +58,24 @@ NVCC_VERSION = '%{cuda_version}'
 def Log(s):
   print('gpus/crosstool: {0}'.format(s))
 
+@contextlib.contextmanager
+def ClosingFileDescriptor(fd):
+  try:
+    yield fd
+  finally:
+    os.close(fd)
+
+
+def NormalizeArgs(args):
+  result = []
+  for arg in args:
```

Comment:
This is not ideal. We invoke this script on every compilation for any C files, thus, on big project, doing this can significantly slow down the build. Any other ways we can do this?

Reply:
When bazel links, it usually bundles the arguments into a file so that the script is invoked with 

```
external/local_config_cuda/crosstool/clang/bin/crosstool_wrapper_driver_is_not_gcc @bazel-out/k8-fastbuild/bin/benchmark/reduce1/benchmark-2.params
```

So to figure out whether `-x cuda` was passed, it needs to go through the arguments file.

But I'm open to tweaking it if you could suggest a better way.

Are you sure the argument normalization would add significantly to the cost? I would expect the process forking to be more expensive.

You could also move your wrapper to native code rather than a Python script if cost is a concern... that's what rust rules does with their wrapper: https://github.com/bazelbuild/rules_rust/tree/main/util/process_wrapper